src/gesture-recognizers.js

```diff
@@ -738,43 +756,78 @@ export class PinchRecognizer extends GestureRecognizer {
   /** @override */
   onTouchMove(e) {
     const {touches} = e;
+    if (!touches || touches.length == 0) {
+      return false;
+    }
     // Pinch touches are not always simultaneous, continue to listen
     // for second touch.
-    if (touches && touches.length == 1) {
+    if (touches.length == 1) {
       return true;
-    } else if (touches && touches.length == 2) {
-      this.lastX1_ = touches[0].clientX;
-      this.lastY1_ = touches[0].clientY;
-      this.lastX2_ = touches[1].clientX;
-      this.lastY2_ = touches[1].clientY;
-      if (this.eventing_) {
-        this.emit_(false, false, e);
-      } else {
-        const dx1 = this.lastX1_ - this.startX1_;
-        const dy1 = this.lastY1_ - this.startY1_;
-        const dx2 = this.lastX2_ - this.startX2_;
-        const dy2 = this.lastY2_ - this.startY2_;
-        // Fingers should move in opposite directions and go over the threshold.
-        if (dx1 * dx2 <= 0 && dy1 * dy2 <= 0) {
-          if (Math.abs(dx1 - dx2) >= PINCH_ACCEPT_THRESHOLD
-            || Math.abs(dy1 - dy2) >= PINCH_ACCEPT_THRESHOLD) {
-            this.signalReady(0);
-          }
-        } else if (Math.abs(dx1 + dx2) >= PINCH_REJECT_THRESHOLD
-          || Math.abs(dy1 + dy2) >= PINCH_REJECT_THRESHOLD) {
-          // Moving in the same direction over a threshold.
-          return false;
-        }
-      }
+    }
+
+    // Have 2+ touches
+    this.lastX1_ = touches[0].clientX;
+    this.lastY1_ = touches[0].clientY;
+    this.lastX2_ = touches[1].clientX;
+    this.lastY2_ = touches[1].clientY;
+
+    // If eventing, always emit gesture with new coordinates
+    if (this.eventing_) {
+      this.emit_(false, false, e);
       return true;
-    } else {
+    }
+
+    // Gesture is 2+ touch but direction indicates not a pinch
+    if (this.pinchRejected_()) {
       return false;
     }
+
+    if (this.pinchReady_()) {
+      this.signalReady(0);
+    }
+    // Pinch gesture detected but threshold not reached, continue listening
+    return true;
+  }
+
+  /**
+   * @return {boolean}
+   * @private
+   */
+  pinchReady_() {
```

Comment:
isPinchReady_?

Reply:
Done and also fixed below. Technically, it's more like "pinchIsReady", "pinchIsRejected". Or maybe since these are private functions of the `PinchRecognizer`, we could directly go with `isReady` and `isRejected`.

Reply:
Talked offline, going to leave as is (`isPinchReady`, `isPinchRejected`).